src/Juvix/Compiler/Pipeline.hs

```diff
@@ -37,11 +37,12 @@ import Juvix.Compiler.Reg.Data.InfoTable qualified as Reg
 import Juvix.Compiler.Reg.Translation.FromAsm qualified as Reg
 import Juvix.Data.Effect.Git
 import Juvix.Data.Effect.Process
+import Juvix.Data.Effect.TaggedLock
 import Juvix.Prelude
 
-type PipelineEff = '[PathResolver, EvalFileEff, Error PackageLoaderError, Error DependencyError, GitClone, Error GitProcessError, Process, Log, Reader EntryPoint, Files, NameIdGen, Builtins, Error JuvixError, HighlightBuilder, Internet, Embed IO]
+type PipelineEff = '[PathResolver, EvalFileEff, Error PackageLoaderError, Error DependencyError, GitClone, Error GitProcessError, Process, Log, TaggedLock, Reader EntryPoint, Files, NameIdGen, Builtins, Error JuvixError, HighlightBuilder, Internet, Embed IO, Resource, Final IO]
```

Comment:
Is it necessary to have both `Embed IO` and `Final IO`?

Reply:
I think so, because of the way we interpret the effects at the top level:

https://github.com/anoma/juvix/blob/c4d349473db82eacb938e8d2e4f456417a4e9361/src/Juvix/Compiler/Pipeline/Run.hs#L57-L75

The IO affects like `runFilesIO` depend on `Embed IO` but `resourceToIOFinal` depends on `Final IO`. We also have `embedToFinal @IO` in the stack of interpreters but for some reason I don't understand this does not mean we can remove the `Embed IO` effect (without doing something else), we get a type error:

<details>

<summary>Type error output, expand for details</summary>

```
src/Juvix/Compiler/Pipeline/Run.hs:75:7: error:
    • Couldn't match type ‘Resource’ with ‘Embed IO’
      Expected: Sem PipelineEff a
                -> Sem
                     '[Juvix.Compiler.Pipeline.Package.Loader.EvalEff.EvalFileEff,
                       Error PackageLoaderError, Error DependencyError,
                       Scoped CloneArgs Git, Error GitProcessError, Process, Log,
                       TaggedLock, Reader EntryPoint, Files, NameIdGen, Builtins,
                       Error JuvixError, HighlightBuilder, Internet, Embed IO, Resource,
                       Final IO]
                     (ResolverState, a)
        Actual: Sem
                  '[PathResolver,
                    Juvix.Compiler.Pipeline.Package.Loader.EvalEff.EvalFileEff,
                    Error PackageLoaderError, Error DependencyError, GitClone,
                    Error GitProcessError, Process, Log, TaggedLock, Reader EntryPoint,
                    Files, NameIdGen, Builtins, Error JuvixError, HighlightBuilder,
                    Internet, Resource, Final IO]
                  a
                -> Sem
                     '[Juvix.Compiler.Pipeline.Package.Loader.EvalEff.EvalFileEff,
                       Error PackageLoaderError, Error DependencyError, GitClone,
                       Error GitProcessError, Process, Log, TaggedLock, Reader EntryPoint,
                       Files, NameIdGen, Builtins, Error JuvixError, HighlightBuilder,
                       Internet, Resource, Final IO]
                     (ResolverState, a)
    • In the second argument of ‘(.)’, namely ‘runPathResolver'’
      In the second argument of ‘(.)’, namely
        ‘runEvalFileEffIO . runPathResolver'’
      In the second argument of ‘(.)’, namely
        ‘mapError (JuvixError @PackageLoaderError)
           . runEvalFileEffIO . runPathResolver'’
   |
75 |     . runPathResolver'
   |       ^^^^^^^^^^^^^^^^
```

</details>

Reply:
I think we could get rid of the `Embed IO` if we refactored the functions that have a constraint like `Member (Embed IO) r` with `MonadIO r`. However, I don't think there is a point of doing that in this pr, so it's fine as it is